### PR TITLE
Add flexibility on td2 to direct packets from CPU to a desired egress queue of front-panel port

### PR DESCRIPTION
**Pull request by wendani**
Signed-off-by: Wenda <wenni@microsoft.com>

**What I did**

Direct CPU packets to a desired egress queue on TD2.

**Why I did it**

Packets from CPU by default go to egress queue 0. Add the flexibility to change the default behavior on td2 to direct CPU packets to use an egress queue # of your need.

Note that on THx, packets from CPU is supposed to go to egress queue 7 of front-panel port . The buffer space associated with queue 7 is reserved specifically for cpu traffic. So on THx, the queue # is hard-coded 7 regardless of what queue index # is fed in the function call. However, we still need the function call in place to direct the cpu traffic to queue 7, because the default sai initialization for THx directs cpu packets to egress queue 0.

**How I verified it**

Build swss_1.0.0_amd64.deb
Install the package in the swss container
Check the egress queue through which protocol packets go:

UC_PERQ_PKT(7).xe0          :                   157                  +1
UC_PERQ_BYTE(7).xe0         :                19,666                +239

**Details if related**

### CODE REVIEW

**Comment by lguohan** [comment]
can you check virtual switch build failure? does this work for broadcom td2, mellanox platform?

**Comment by wendani** [comment]
On td2, packets from CPU are correctly directed to egress queue 7

**Comment by wendani** [comment]
I checked the error. ValueError: arp_update                       STOPPED   Not started
But when I shift to the original swss package, I see arp_update is also stopped inside the swss container.

root@str-a7050-acs-1:/# supervisorctl 
arp_update                       STOPPED   Not started

**Comment by wendani** [comment]
retest this please

**Comment by wendani** [comment]
💯

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -1869,6 +1870,22 @@ bool PortsOrch::addHostIntfs(Port &port, string alias, sai_object_id_t &host_int
     strncpy((char *)&attr.value.chardata, alias.c_str(), SAI_HOSTIF_NAME_SIZE);
     attrs.push_back(attr);
 
+    const char *platform = getenv("platform");
+    if (platform && strstr(platform, BRCM_PLATFORM_SUBSTRING))
```
Comment (nikos-li) [inline orchagent/portsorch.cpp]: Platform specific checks are not allowed in PI code. There was a lengthy discussion last year and no more new checks are allowed other than the existing ones.

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -33,6 +33,7 @@ extern CrmOrch *gCrmOrch;
 #define DEFAULT_VLAN_ID     1
 #define FLEX_STAT_COUNTER_POLL_MSECS "1000"
 #define STAT_COUNTER_FLEX_COUNTER_GROUP "STAT_COUNTER"
+#define CPU_TRAF_QUEUE_IDX  0
```
[the remark this answers is not in the document]
Comment (wendani) [reply orchagent/portsorch.cpp]: Hi, the description is confusing. Let me know if you can understand clearly.

**Pull request closed without merging**